web-search-engines/langchain4j-web-search-engine-searchapi/src/test/java/dev/langchain4j/web/search/searchapi/SearchApiWebSearchEngineIT.java

```diff
@@ -0,0 +1,336 @@
+		assertThat(searchParams.metadata()).containsEntry("language", "en");
+		assertThat(searchParams.metadata()).containsEntry("safe_search", "strict");
+		assertThat(searchParams.metadata()).containsEntry("page", "1");
+
+		// then
+		Map<String, Object> searchMetadata = webSearchResults.searchMetadata();
+		assertThat(searchMetadata).containsKey("id");
+		assertThat(searchMetadata.get("id")).isNotNull();
+		assertThat(searchMetadata).containsKey("created_at");
+		assertThat(searchMetadata.get("created_at")).isNotNull();
+		assertThat(searchMetadata).containsKey("request_url");
+		assertThat(searchMetadata.get("request_url")).isNotNull();
+
+		// then
+		List<WebSearchOrganicResult> results = webSearchResults.results();
+		
+		assertThat(results).isNotEmpty();
+
+		results.forEach(result -> {
+			assertThat(result.title()).isNotBlank();
+			assertThat(result.url()).isNotNull();
+			assertThat(result.snippet()).isNotBlank();
+			assertThat(result.content()).isNull();
+			assertThat(result.metadata()).isNotNull();
+		});
+	}
+
+	@Test
+	void customize_baidu_search_parameters() {
+		// given
+		SearchApiWebSearchEngine searchapiWebSearchEngine = SearchApiWebSearchEngine.builder()
+				.apiKey(System.getenv(DEFAULT_ENV_VAR))
+				.engine("baidu")
+				.logRequests(true)
+				.customizeParametersFunc(
+						(params) -> {
+							params.put("ct", "0");
+							params.put("num", "5");
+							params.put("page", "1");
+						})
+				.build();
+
+		// when
+		WebSearchResults webSearchResults = searchapiWebSearchEngine.search("chatgpt");
+
+		// then
+		assertThat(webSearchResults.searchInformation().totalResults()).isPositive();
+		assertThat(webSearchResults.searchInformation().pageNumber()).isEqualTo(1);
+
+		// then
+		WebSearchInformationResult searchParams = webSearchResults.searchInformation();
+		assertThat(searchParams.metadata()).containsEntry("engine", "baidu");
+		assertThat(searchParams.metadata()).containsEntry("q", "chatgpt");
+		assertThat(searchParams.metadata()).containsEntry("ct", "0");
+		assertThat(searchParams.metadata()).containsEntry("num", "5");
+		assertThat(searchParams.metadata()).containsEntry("page", "1");
+
+		// then
+		Map<String, Object> searchMetadata = webSearchResults.searchMetadata();
+		assertThat(searchMetadata).containsKey("id");
+		assertThat(searchMetadata.get("id")).isNotNull();
+		assertThat(searchMetadata).containsKey("created_at");
+		assertThat(searchMetadata.get("created_at")).isNotNull();
+		assertThat(searchMetadata).containsKey("request_url");
+		assertThat(searchMetadata.get("request_url")).isNotNull();
+
+		// then
+		List<WebSearchOrganicResult> results = webSearchResults.results();
+		
+		assertThat(results).isNotEmpty();
+
+		results.forEach(result -> {
+			assertThat(result.title()).isNotBlank();
+			assertThat(result.url()).isNotNull();
+			assertThat(result.snippet()).isNotNull();
+			assertThat(result.content()).isNull();
+			assertThat(result.metadata()).isNotNull();
+		});
+	}
+	
+//	@Test
```

Comment:
could you remove this commented test? If it's not implemented yet it's not needed.